internal/ecgit/ecgit.go

```diff
@@ -0,0 +1,306 @@
+	"github.com/spf13/afero"
+	giturls "github.com/whilp/git-urls"
+	"golang.org/x/oauth2"
+
+	"github.com/hacbs-contract/ec-cli/internal/utils"
+)
+
+var CloneRepoWithAuth = cloneRepoWithAuth
+var CreateAutomatedPR = createAutomatedPR
+var CreateBranch = createBranch
+var CreatePullRequest = createPullRequest
+var ApplyDiff = applyDiff
+var CommitChange = commitChange
+var PushChange = pushChange
+var NewGithubClient = newGithubClient
+
+// cloneRepoWithAuth clones the specified repository.
+func cloneRepoWithAuth(repoURL string, destDir string, ctx context.Context) (*git.Repository, error) {
+	log.Info("No private key file provided, using default authentication")
+	if os.Getenv("GITHUB_USERNAME") == "" {
+		log.Debug("Unable to clone repository, GITHUB_USERNAME not set")
+		return nil, errors.New("GITHUB_USERNAME not set")
+	}
+	if os.Getenv("GITHUB_TOKEN") == "" {
+		return nil, errors.New("GITHUB_TOKEN not set")
+	}
+	auth := &http.BasicAuth{
+		Username: os.Getenv("GITHUB_USERNAME"),
+		Password: os.Getenv("GITHUB_TOKEN"),
+	}
+
+	r, err := git.PlainCloneContext(ctx, destDir, false, &git.CloneOptions{
+		Auth:     auth,
+		URL:      repoURL,
+		Progress: os.Stdout,
+	})
+	if err != nil {
+		log.Debug("Unable to clone repository due to error")
+		return nil, err
+	}
+	return r, nil
+}
+
+// createBranch creates a new branch on the specified repository.
+func createBranch(repo *git.Repository, branchName string) (*git.Worktree, error) {
+	// Create a new branch
+	headRef, err := repo.Head()
+	if err != nil {
+		log.Debug("Unable to create branch due to getting HEAD ref")
+		return nil, err
+	}
+	refName := plumbing.NewBranchReferenceName(branchName)
+	ref := plumbing.NewHashReference(refName, headRef.Hash())
+	err = repo.Storer.SetReference(ref)
+	if err != nil {
+		log.Debugf("Error setting ref: %s\n", ref)
+		return nil, err
+	}
+	w, err := repo.Worktree()
+	if err != nil {
+		log.Debug("Unable to get repo worktree")
+		return nil, err
+	}
+	err = w.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.ReferenceName(refName.String()),
+	})
+	if err != nil {
+		log.Debugf("Error checking out branch: %s\n", refName.String())
+		return nil, err
+	}
+	return w, nil
+}
+
+// applyDiff applies the diff to the specified repository.
+func applyDiff(repoPath string, diffFilePath string) error {
+	err := exec.Command("git", "-C", repoPath, "apply", diffFilePath).Run()
+	if err != nil {
+		log.Debugf("Error applying diff file '%s'\n", diffFilePath)
+		return err
+	}
+	return nil
+}
+
+// commitChange commits the changes in the worktree to the specified repository.
+func commitChange(repo *git.Repository, refName string, message string) (*object.Commit, error) {
+	w, err := repo.Worktree()
+	if err != nil {
+		log.Debug("Error getting repo worktree")
+		return nil, err
+	}
+	_, err = w.Add(".")
+	if err != nil {
+		log.Debug("Error adding files")
+		return nil, err
+	}
+	commit, err := w.Commit(message, &git.CommitOptions{
+		Author: &object.Signature{
```

Comment:
Would it be best to just rely on whatever .gitconfig is available? (Should be done automatically by go-git)